podcasts/AutoplayHelper.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+class AutoplayHelper {
+    static let shared = AutoplayHelper()
+
+    private let userDefaults: UserDefaults
+    private let userDefaultsKey = "playlist"
+
+    var lastPlaylist: EpisodesDataManager.Playlist? {
+        if let rawDictionary = userDefaults.dictionary(forKey: userDefaultsKey),
+           let dictData = try? JSONSerialization.data(withJSONObject: rawDictionary) {
+            return try? JSONDecoder().decode(EpisodesDataManager.Playlist.self, from: dictData)
+        }
+
+        return nil
```

Comment:
If we switch to saving the data directly (mentioned below) this can be refactored down to:

```suggestion
        userDefaults.data(forKey: userDefaultsKey).flatMap {
            try? JSONDecoder().decode(EpisodesDataManager.Playlist.self, from: $0)
        }
```